src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -432,4 +443,27 @@ protected function getPreviewCombinationProduct(CombinationId $combinationId): ?
 
         return null;
     }
+
+    public function getAssociatedShopIds(ImageId $imageId): array
```

Comment:
the param is not important, but return type annotation is, cuz it tells us whats inside the array
```suggestion
    /**
     * @param ImageId $imageId
     * @return ShopId[]
    /*
    public function getAssociatedShopIds(ImageId $imageId): array
```